src/filesink.cpp

```diff
@@ -10,6 +10,7 @@
 #include "filesinkhelper.ipp"
 #include <cassert>
 #include <chrono>
+#include <filesystem>
```

Comment:
dang, apparently c++-17 still has the `experimental/filesystem` on gcc (Linux/OSX). I think that's why it's failing. 

https://stackoverflow.com/questions/45867379/why-does-gcc-not-seem-to-have-the-filesystem-standard-library

Reply:
We use this snippet of code to make getting to std::filesystem work on a variety of compilers/platforms. Not exhaustive, of course.

```suggestion
#if defined(_WIN32)
#    if _MSC_VER >= 1914
#        include <filesystem>
namespace fs = std::filesystem;
#    else
#        include <experimental/filesystem>
namespace fs = std::experimental::filesystem;
#    endif
#elif defined(__GNUC__) && __GNUC__ <= 7
#    include <experimental/filesystem>
namespace fs = std::experimental::filesystem;
#elif defined(__GNUC__) && __GNUC__ >= 8
#    include <filesystem>
namespace fs = std::filesystem;
#elif defined(__clang__) && __clang_major__ <= 8
#    include <experimental/filesystem>
namespace fs = std::experimental::filesystem;
#elif defined(__clang__) && __clang_major__ <= 10
#    include <filesystem>
namespace fs = std::filesystem;
#endif
```

Reply:
I wonder if this could be used?
Not sure on Windows but I think it works on Linux and OSX
```
// since C++ 20
#include <version>

#ifdef __cpp_lib_filesystem
    #include <filesystem>
    using fs = std::filesystem;
#elif __cpp_lib_experimental_filesystem
    #include <experimental/filesystem>
    using fs = std::experimental::filesystem;
#else
    #error "no filesystem support ='("
#endif
```

Reply:
Or use https://github.com/gulrak/filesystem?

Reply:
I think having just made the effort get onto c++17 it makes more sense to start using it.